### PR TITLE
Devise2

### PR DESCRIPTION
**Pull request by leherreras**
(5) As a user I should be able to sign up so my account is created in Target MVD
(2) As a user I should be able to logout, so I can no longer be logged in
(3) As a user I should be able to login so I can use my Target MVD account
(3) As a user I should be able to confirm my account after signing up in order to ensure that all the user's email are valid
(3) As a user I should be able to reset my password so I can regain access to my account if I forgot it

### CODE REVIEW

File: spec/requests/api/v1/registrations_spec.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  let(:valid_attributes) do
+    { email: 'test@test.com', gender: 'male', password: '123456', password_confirmation: '123456' }
+  end
+
+  let(:invalid_attributes) do
+    { email: 'test@test.com', gender: 'male_bat', password: '123456', password_confirmation: '123456' }
+  end
+
+  describe 'POST /create' do
+    context 'with valid parameters' do
```
Comment (enzofab91) [inline spec/requests/api/v1/registrations_spec.rb:21]: Nice. Add another test that validates the user information receveived is actually the data store in the database.

**Review by enzofab91 — APPROVED** [review]
Good work! 👏 

Check the linters suggestions.

**Review by sebastiancaraballo — COMMENTED** [review]
I'd suggest smaller PRs

File: app/controllers/api/v1/passwords_controller.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
+module Api
+  module V1
+    class PasswordsController < DeviseTokenAuth::PasswordsController
+      def edit
+        # if a user is not found, return nil
```
Comment (antoalconti) [inline app/controllers/api/v1/passwords_controller.rb:7]: I think that all this actions are executed by devise token auth, i suggest not redefine this edit

File: config/locales/en.yml [hunk 3]
```diff
@@ -1,33 +1,14 @@
...
-# available at https://guides.rubyonrails.org/i18n.html.
-
 en:
   hello: "Hello world"
+
+  api:
+    errors:
+      invalid_content_type: Invalid content type header
+      missing_param: A required param is missing
+      not_found: Couldn't find the record
```
Comment (antoalconti) [inline config/locales/en.yml]:
```suggestion
      not_found: "Couldn't find the record"
```

Comment (antoalconti) [reply config/locales/en.yml]: same in the followings en to include the '

File: spec/requests/api/v1/registrations_spec.rb [hunk 4]
```diff
@@ -0,0 +1,50 @@
+require 'rails_helper'
+
+RSpec.describe 'Api::V1::Registrations', type: :request do
+
+  let(:email)                 { 'test@test.com' }
+  let(:password)              { '123456' }
```
Comment (antoalconti) [inline spec/requests/api/v1/registrations_spec.rb:7]: i recommend you install factory bot gem, it will help you to create data for rspec

**Review by antoalconti — COMMENTED** [review]
great work! Same than seba, i recommend create smaller pr

File: app/controllers/api/v1/passwords_controller.rb (hunk 2 above)
Comment (leherreras) [reply app/controllers/api/v1/passwords_controller.rb:7]: The idea in this "edit" is only return a Json, but the functionality is the same.

Comment (antoalconti) [reply app/controllers/api/v1/passwords_controller.rb:7]: Make senses

**Pull request closed without merging**